src/commands/base-command.ts

```diff
@@ -248,13 +282,24 @@ export abstract class BaseCliCommand<T extends ICommandArgs> {
             ConsoleUtil.colorizeLogs = false;
         }
 
-        await AwsUtil.InitializeWithProfile(command.profile);
-
+        await AwsUtil.InitializeWithProfile(command.profile, command.govCloud);
 
         if (command.masterAccountId !== undefined) {
             AwsUtil.SetMasterAccountId(command.masterAccountId);
         }
 
+        if (command.govCloudProfile !== undefined) {
```

Comment:
i think i would prefer a single function to be exposed by AwsUtil that would store a class that looks something like:

```
class PartitionConfig {
   Partition: string  = 'us-gov';
   Region: string  = 'us-gov-east-1';
   Profile: string  = 'profile name';
   ManagementAccountId?: string;  //will be set/lazyloaded by some other code
}

//called when the intention is to mirror commercial -> partition
AwsUtil.SetMirrorToPartition(config: PartitionConfig) => { 
  AwsUtil.PartititionConfig = config;
  AwsUtil.MirrorToPartition = true; // default would be false
}

//called when running tasks in non-commercial partition
AwsUtil.SetPartitionConfig(config: PartitionConfig) => { 
  AwsUtil.PartititionConfig = config;
  AwsUtil.Partition = config.Partition; // default would be 'commercial'
}

```
its very annoying for me to paste pseudo code here, the benefit i would be seeking is:
- less methods that need to be called to set variables (there is two scenario's, each get their function)
- grouping some of the required values in a single type/variable. this makes it easier to get all relevant values in a single var.

Reply:
I'd probably have to circle back to this